gitclone/checkout.go

```diff
@@ -333,6 +331,26 @@ func selectFilterTreeFetchOption(method CheckoutMethod, opts fetchOptions, filte
 	}
 }
 
+func idealDefaultCloneDepth(method CheckoutMethod) int {
+	const defaultCloneDepth = 50
+
+	switch method {
+	case CheckoutNoneMethod,
+		CheckoutPRMergeBranchMethod,
+		CheckoutPRManualMergeMethod,
```

Comment:
I don't think it's correct. Manual merging requires the commit history, this is the case that gives the "refusing to merge unrelated histories" on shallow checkouts.